lib/pacemaker/pcmk_status.c

```diff
@@ -137,6 +146,7 @@ pcmk__output_cluster_status(pcmk__output_t *out, stonith_t *stonith, cib_t *cib,
     stonith_history_free(stonith_history);
     stonith_history = NULL;
     pe_free_working_set(data_set);
+    free(our_nodename);
```

Comment:
Freeing it every call defeats the purpose of making it static. (And freeing it without nulling it means the next call will be a use-after-free.) We can either skip the free and let it leak, or get the node name in crm_mon and pass it as an argument here, or make it a static global in libcrmcommon freed at crm_exit().

Reply:
Whoops, I'm glad you caught this. I don't even remember adding the frees... I must have been on cleanup-autopilot.

Reply:
I think it's fine to let it "leak." We do that in `get_local_node_name()`. With that said, I like the idea of storing it as lib-global if no complications arise.

Reply:
Updated. We might be able to integrate this in other places too (e.g., `fsa_our_uname`; try `pcmk__our_uname` for `stonith_our_uname` until it's deprecated and then replace the variable name; maybe others). I haven't looked closely at those.